test/tools/IsolTestOptions.cpp

```diff
@@ -59,11 +59,20 @@ std::string editorPath()
 IsolTestOptions::IsolTestOptions(std::string* _editor):
 	CommonOptions(description)
 {
+	editor = _editor;
```

Comment:
The fact that this function gets a pointer to a string stored outside is really weird. I'd expect it to be stored in the struct.

My suggestion would be to make `editor` just a string and make the caller just take it out of the `IsolTestOptions` object.

Reply:
I agree that it is weird, but I didn't create the interface that led to that behavior. The `editor` string is used by isoltest.cpp for executing editor commands, so I would assume that is why it controls the lifecycle. `IsolTestOptions` really only needs it to initialize the `options_description` object but since that can't be done in the constructor and still allow the option defaults to be modified, it's just passed through a helper variable right now rather than introducing a dependency to get it directly.

Reply:
I know, it was here before your changes, I'm not saying that this is wrong on your part :)  I'm just saying that it's ok to refactor it instead of extending this design. I think it should be changed now because storing the pointer in the struct is worse than just passing it via parameter. The struct could easily outlive the string.

Reply:
I understand. However, `isoltest` needs the `edtior` resource independently of whether or not the idea of the `IsolTestOptions` class even existed.  How would you suggest avoiding the reverse problem if the resource is moved? `IsolTestOptions` is not necessarily created before the editor string is used by `isoltest`. An `IsolTestOptions` object cannot exist without a `TestTool` object so the resource `IsolTestOptions` points to is guaranteed to exist and the non-static `IsolTestOptions` object is stored as a member of `TestTool` alongside the static `editor` variable so it will necessarily be destroyed at least simultaneously to the resource it points to if not before.

Reply:
> `isoltest` needs the `edtior` resource independently of whether or not the idea of the `IsolTestOptions` class even existed.

I don't think it does. The static `editor` variable is used only by `TestTool::handleResponse()`, which is not a static function and could just take it from `m_options.editor` instead (if we make it a `string` rather than a pointer).

We don't even have to move it - I originally suggested that because I did not analyze the whole logic here and I thought that the owner of the `editor` variable is actually the one creating the options struct.

Reply:
I didn't fully follow the dataflow so I missed that as well. What you said makes sense and has been changed.